filebrowser_safe/views.py

```diff
@@ -127,8 +127,12 @@ def browse(request):
 
         # FILTER / SEARCH
         append = False
-        if fileobject.filetype == request.GET.get('filter_type', fileobject.filetype) and get_filterdate(request.GET.get('filter_date', ''), fileobject.date):
-            append = True
+        if fileobject.filetype == request.GET.get('filter_type', fileobject.filetype):
+            filter_date = request.GET.get('filter_date', '')
+            if filter_date and get_filterdate(filter_date, fileobject.date):
+                append = True
+            if not filter_date:
```

Comment:
This clause is not obvious, but if you look in `get_filterdate` you see that `''` leads to a truthy return.